### ISSUE TITLE
[pdf]: undefined sequence for `RedeclareSectionCommand`

### ISSUE THREAD

**Issue by juliantao** [quarto-dev/quarto-cli#657]
Using the dev version, LaTex refuses to compile if a custom section format is defined in the header.

The following won't compile:

```markdown
---
title: RedeclareSectionCommand
format: pdf 
header-includes: |
        \RedeclareSectionCommand[
          beforeskip=-6pt plus 1pt minus 1pt,
          afterskip=1sp plus 1sp minus 1sp
        ]{section}
---

# Section

This is a section.
```

If the custom definition is removed, it compiles.

```markdown
---
title: RedeclareSectionCommand
format: pdf 
---

# Section

This is a section.
```

**Comment by jjallaire** [comment]
To make sure that this is valid LaTeX in a pandoc LaTeX document you might want to try `format: latex` and then compiling the LaTeX standalone (e.g. w/ `pdflatex` or `mkindex`) If that yields the same error then there is something not compatible about that particular fragment alongside the standard pandoc LaTeX

Note that we did just make a large number of changes to how we handle Pandoc LaTeX templates so it would also be instructive to create a diff between the output produced today and that of e.g one week ago.

**Comment by juliantao** [comment]
Sorry that I was in a rush last night and did not investigate the reason in the details.

It turns out that the default `documentclass` changed from `scrartcl` to `article`, possibly due to the recent work on templates. 

Tex file from the most recent dev version,

```tex
\documentclass[
]{article}\usepackage{amsmath,amssymb}
```

From the version with commit [`73a6e3f`](https://github.com/quarto-dev/quarto-cli/commit/73a6e3fa7d354f6dea6bde0fd2dc9ef67f99b44b) (before the template related commits)

```tex
\documentclass[
  letterpaper,
  DIV=11,
  numbers=noendperiod]{scrartcl}
\usepackage{amsmath,amssymb}
```

The custom section format in my MRE follows that of KOMA-scripts. After setting the `documentclass` to `scrartcl` directly in the `yml` header, it's working now.

```markdown
---
title: RedeclareSectionCommand
format: 
  pdf:
    documentclass: scrartcl
    papersize: letter
    keep-tex: true
    header-includes: |
            \RedeclareSectionCommand[
              beforeskip=-6pt plus 1pt minus 1pt,
              afterskip=1sp plus 1sp minus 1sp
            ]{section}
---

# Section

This is a section.
```

So, is it the final decision that the default `documentclass` will be changed to `article` for Quarto pdf format? Or is it oversight of some sort?

If it is the former, I may need to update all my quarto files with a pdf format...

**Comment by jjallaire** [comment]
No, that's an oversight. We'll investigate and remedy ASAP (thanks for reporting this!)

**Comment by jjallaire** [comment]
Okay, this is fixed here: https://github.com/quarto-dev/quarto-cli/commit/3edc65421221dd46dc2539512e53619edf85f9e2

Fix is available in v0.9.269 release (https://github.com/quarto-dev/quarto-cli/releases/tag/v0.9.269)
